File: articles/opportunity-recognition-and-evaluation/c/opportunity_score.c
/*
Low-level opportunity score utilities.
Compile: cc opportunity_score.c -o opportunity_score
Run: ./opportunity_score
*/

#include <stdio.h>

typedef struct {
    const char *name;
    double signal;
    double capability;
    double desirability;
    double viability;
    double timing;
    double learning;
    double option_value;
    double strategic_fit;
    double ethics;
    double risk;
    double confidence;
} OpportunityProfile;

double profile_score(OpportunityProfile o) {
    return 0.13 * o.signal +
           0.14 * o.capability +
           0.12 * o.desirability +
           0.12 * o.viability +
           0.10 * o.timing +
           0.12 * o.learning +
           0.11 * o.option_value +
           0.10 * o.strategic_fit +
           0.10 * o.ethics -
           0.14 * o.risk;
}

int main(void) {
    OpportunityProfile opportunities[] = {
        {"Emerging Technology Adjacency", 0.74, 0.78, 0.72, 0.71, 0.76, 0.70, 0.66, 0.78, 0.62, 0.46, 0.68},
        {"High-Hype Weak-Fit Opportunity", 0.86, 0.31, 0.77, 0.39, 0.48, 0.62, 0.44, 0.40, 0.42, 0.78, 0.38},
        {"Slow-Build Sustainability Opportunity", 0.62, 0.73, 0.81, 0.74, 0.67, 0.64, 0.72, 0.76, 0.82, 0.41, 0.66}
    };

    int count = sizeof(opportunities) / sizeof(opportunities[0]);

    for (int i = 0; i < count; i++) {
        double score = profile_score(opportunities[i]);
        printf("%s | score %.3f | confidence-adjusted %.3f\n",
               opportunities[i].name,
               score,
               score * opportunities[i].confidence);
    }

    return 0;
}

File: articles/opportunity-recognition-and-evaluation/rust/opportunity_score.rs
// Command-line opportunity profile scoring scaffold.
// Compile: rustc opportunity_score.rs -o opportunity_score
// Run: ./opportunity_score

#[derive(Debug)]
struct OpportunityProfile {
    name: &'static str,
    signal: f64,
    capability: f64,
    desirability: f64,
    viability: f64,
    timing: f64,
    learning: f64,
    option_value: f64,
    strategic_fit: f64,
    ethics: f64,
    risk: f64,
    confidence: f64,
}

fn profile_score(o: &OpportunityProfile) -> f64 {
    0.13 * o.signal
        + 0.14 * o.capability
        + 0.12 * o.desirability
        + 0.12 * o.viability
        + 0.10 * o.timing
        + 0.12 * o.learning
        + 0.11 * o.option_value
        + 0.10 * o.strategic_fit
        + 0.10 * o.ethics
        - 0.14 * o.risk
}

fn main() {
    let opportunities = vec![
        OpportunityProfile { name: "Emerging Technology Adjacency", signal: 0.74, capability: 0.78, desirability: 0.72, viability: 0.71, timing: 0.76, learning: 0.70, option_value: 0.66, strategic_fit: 0.78, ethics: 0.62, risk: 0.46, confidence: 0.68 },
        OpportunityProfile { name: "High-Hype Weak-Fit Opportunity", signal: 0.86, capability: 0.31, desirability: 0.77, viability: 0.39, timing: 0.48, learning: 0.62, option_value: 0.44, strategic_fit: 0.40, ethics: 0.42, risk: 0.78, confidence: 0.38 },
        OpportunityProfile { name: "Slow-Build Sustainability Opportunity", signal: 0.62, capability: 0.73, desirability: 0.81, viability: 0.74, timing: 0.67, learning: 0.64, option_value: 0.72, strategic_fit: 0.76, ethics: 0.82, risk: 0.41, confidence: 0.66 },
    ];

    for o in opportunities {
        let score = profile_score(&o);
        println!("{} | score {:.3} | confidence-adjusted {:.3}", o.name, score, score * o.confidence);
    }
}
